### PR TITLE
[TensorFlow][ReRelease] TF2.9 Re-Release

### PR DESCRIPTION
**Pull request by ohadkatz**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description

This PR is to re-release TF2.9 , along with updating sagemaker debugger to latest version

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ X ] Revision A: `sagemaker_remote_tests = "standard"`

## PR Checklist
- [ X ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ X ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ohadkatz** [comment]
Tests ran on [c47793](https://github.com/aws/deep-learning-containers/pull/2603/commits/7c477931cfddad85890916893a9f4968b4e90db6)

**Comment by ohadkatz** [comment]
ECS/EKS/EC2 Tests Ran on [897ea11](https://github.com/aws/deep-learning-containers/pull/2603/commits/897ea11f48d426d9f4ead011a48746afa6f25df2)

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -263,14 +263,15 @@ RUN HOROVOD_WITH_MPI=1 \
    && ${PIP} install --no-cache-dir horovod==0.24.2
 
 RUN $PYTHON -m pip install --no-cache-dir -U \
+    numba \
```
Comment (kace) [inline tensorflow/training/docker/2.9/py3/Dockerfile.cpu:266]: Curious to know why we had a version constraint for `numba` in the first place?

File: tensorflow/buildspec-2-9.yml [hunk 2]
```diff
@@ -41,149 +41,149 @@ context:
...
+    image_size_baseline: &IMAGE_SIZE_BASELINE 9307
     device_type: &DEVICE_TYPE gpu
     python_version: &DOCKER_PYTHON_VERSION py3
     tag_python_version: &TAG_PYTHON_VERSION py39
     cuda_version: &CUDA_VERSION cu112
     os_version: &OS_VERSION ubuntu20.04
-    tag: !join [ *FRAMEWORK_VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION, "-", *OS_VERSION, "-ec2" ]
-    docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
+    tag: !join [ *VERSION, "-", *DEVICE_TYPE, "-", *TAG_PYTHON_VERSION, "-", *CUDA_VERSION,
+      "-", *OS_VERSION, "-ec2" ]
```
Comment (kace) [inline tensorflow/buildspec-2-9.yml:70]: nit: Was this whitespace intentional?

File: tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu [hunk 3]
```diff
@@ -405,6 +405,9 @@ ENV TF_AUTOTUNE_THRESHOLD=2
 
 ENV LD_LIBRARY_PATH="/usr/local/lib/${PYTHON}/site-packages/smdistributed/dataparallel/lib:$LD_LIBRARY_PATH"
 
+# Remove python kernel installed by sparkmagic
```
Comment (kace) [inline tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:408]: Was there a specific reason this was added now?

Comment (ohadkatz) [reply tensorflow/training/docker/2.9/py3/cu112/Dockerfile.gpu:408]: Removing ipykernel to ensure proper usage in SM Studio

File: tensorflow/buildspec-2-9.yml (hunk 2 above)
Comment (ohadkatz) [reply tensorflow/buildspec-2-9.yml:70]: I formatted the YML file before uploading

File: tensorflow/training/docker/2.9/py3/Dockerfile.cpu (hunk 1 above)
Comment (ohadkatz) [reply tensorflow/training/docker/2.9/py3/Dockerfile.cpu:266]: Might have been a constraint on the numpy version available!

**Review by kace — APPROVED** [review]

**Pull request merged**